### PR TITLE
rollback commits and tags when publish fails

### PR DESCRIPTION
**Pull request by stropitek**
@targos looks good to you?

#199

### CODE REVIEW

File: bin/cheminfo-publish.js [hunk 1]
```diff
@@ -164,7 +164,16 @@ You chose ${formatToBump(bump)} instead.`);
...
     // Publish package
     console.log('Publishing package');
-    log(yield execNpm('publish'));
+    var publishOutput;
+    try {
+        publishOutput = yield execNpm('publish');
+    } catch(e) {
+        console.error('npm publish failed, rolling back commits and tags');
+        yield child_process.exec(`git tag -d v${newVersion}`);
+        yield child_process.exec('git reset --hard HEAD~1');
```
Comment (targos) [inline bin/cheminfo-publish.js:173]: `git reset --hard origin/master` would be safer

**Review by targos — APPROVED** [review]

**Comment by targos** [comment]
please test it with `cheminfo-js/test` when it's deployed

File: bin/cheminfo-publish.js (hunk 1 above)
Comment (stropitek) [reply bin/cheminfo-publish.js:173]: Wouldn't that make you loose commits that you have forgotten to push to origin? You would need to push changes earlier in the script then.

The `reset HEAD~1` can only be executed after the `commit` a little bit higher in the publish script, so I think it is safe

Comment (targos) [reply bin/cheminfo-publish.js:173]: Ok

**Pull request merged**